### ISSUE TITLE
toObject() transform causes *lost data* during save.

### ISSUE THREAD

**Issue by heisian** [Automattic/mongoose#5807]
```node
/**
 * BUG REPORT: `toObject()` transform causes lost data during save.
 */

const mongoose = require('mongoose');


/** Embedded Schema */
const MyEmbeddedSchema = mongoose.Schema(
  {
    someMoreData: String,
  },
);


/**
 * This plugin filters out the `someMoreData` property
 * when calling `toObject()` or `toJSON()` on MyEmbeddedSchema.
 */
MyEmbeddedSchema.plugin(schema => {
  const filterSomeMoreData = (doc, returnDoc) => {
    delete returnDoc.someMoreData;
    return returnDoc;
  };
  // The use case for this is hiding sensitive data from the client
  schema.options.toJSON = { transform: filterSomeMoreData };
  schema.options.toObject = { transform: filterSomeMoreData };
});


/** Base Schema */
const MySchema = mongoose.Schema(
  {
    embedded: MyEmbeddedSchema,
  }
);

const Model = mongoose.model('MODEL', MySchema);


/** Application Logic */
(async () => {
  await new Promise(resolve => {
    mongoose.Promise = global.Promise;
    mongoose.connect('mongodb://127.0.0.1/mytestdb', { useMongoClient: true });
    mongoose.connection.once('open', () => {
      resolve(mongoose.connection);
    });
  });

  const model = new Model();

  // Commenting this out results in `someMoreData` saving as expected
  await model.save();

  model.embedded = {
    someMoreData: 'something',
  };

  await model.save();

  const fetched = await Model.findById(model._id).exec();

  process.stdout.write(`Should log 'something': ${fetched.embedded.someMoreData}`);

  await mongoose.connection.close();
})();
```

The output of this script is:
```bash
$ node BUG.js
Should log `something`: undefined
```

**Comment by wlingke** [comment]
This is a legit issue.

**Comment by vkarpov15** [comment]
Thanks for reporting, will fix ASAP

**Comment by vkarpov15** [comment]
Fixed in master, fix will be in 4.13.3. Thanks again for the detailed repro :+1:

**Comment by heisian** [comment]
thank you for the incredibly quick fix!
